nuxeo-core/nuxeo-core-management/src/main/java/org/nuxeo/ecm/core/management/statuses/StatusServlet.java

```diff
@@ -128,6 +152,21 @@ private boolean isStarted() {
     @Override
     public void init() throws ServletException {
         log.debug("Ready.");
+        pm = Framework.getService(ProbeManager.class);
     }
 
+    protected HealthCheckResult getOrRunHealthCheck(String probe) {
+        if (StringUtils.isEmpty(probe)) { // run all healthCheck probes
+            return getProbeManager().getOrRunHealthChecks();
+        }
+        return getProbeManager().getOrRunHealthCheck(probe);
+
+    }
+
+    protected ProbeManager getProbeManager() {
+        if (pm == null) {
+            pm = Framework.getService(ProbeManager.class);
```

Comment:
No, don't store the result of `Framework.getService` in a field of a long-lived object, it prevents hot-reload from working correctly. Just look it up when needed.

What I meant was:

    protected HealthCheckResult getOrRunHealthCheck(String probe) {
        ProveManager pm = Framework.getService(ProbeManager.class);
        if (StringUtils.isEmpty(probe)) { // run all healthCheck probes
            return pm.getOrRunHealthChecks();
        } else {
            return pm.getOrRunHealthCheck(probe);
        }
    }